tree/ntuple/v7/src/RFieldVisitor.cxx

```diff
@@ -94,10 +98,195 @@ void ROOT::Experimental::RPrepareVisitor::VisitField(const Detail::RFieldBase &/
       fDeepestLevel = level;
 }
 
+//------------------------ RValueVisitor --------------------------------
+
+void ROOT::Experimental::RValueVisitor::VisitField(const Detail::RFieldBase &field, int level)
+{
+   for (int i = 0; i < level; ++i) fOutput << "  ";
+   fOutput << "\"" << field.GetName() << "\": ";
+   
+   // A custom object (has ENTupleStructure::kRecord) should let its subfields display
+   // its entries instead of displaying all by itself.
+   // So the field of the custom object should only display its name and make an early quit here.
+   if (field.GetStructure() == ENTupleStructure::kRecord) {
+      fOutput << std::endl;
+      return;
+   }
+   
+   // const_cast may lead to undefined behaviour, so this procedure of obtaining
+   // the entries is only used when there are no good alternatives.
+   Detail::RFieldBase* fieldPtr = const_cast<Detail::RFieldBase*>(&field);
+   Detail::RFieldValue value{fieldPtr->GenerateValue()};
```

Comment:
I guess you need the `const_cast` for `GenerateValue()`.  Can `GenerateValue()` become a const method?

Reply:
Yes, const_cast is needed for `GenerateValue()` but it can't become a const function. Setting it const requires changing following members as const: 
`RField::GenerateValue() const` 
->` RField::GenerateValue(void*) const `
   -> `template <typename... ArgsT>RField::GenerateValue(void*, ArgsT&&... args)` 
      -> `RFieldValue::const fField`
          -> `RFieldValue::GetField()` (should now return `const Detail::RFieldBase*`)
              -> `RField::Append(const reference RFieldValue) const`
                  -> `RField::DoAppend( const ref RFieldValue) const `-> doesn't because members such as `fNWritten` are updated

Reply:
All right, let's leave the const_cast

Reply:
I'm currently rewriting almost everything related to std::vector and std::array to get rid of the const_cast and gInterpreter. I hope to get it finished by tomorrow noon...